java/custom/javax/microedition/lcdui/Graphics.java

```diff
@@ -0,0 +1,1559 @@
+ * <em>Computer Graphics V18 N3 (SIGGRAPH 1984)</em>, p. 253-259.
+ * </dl>
+ *
+ * @since MIDP 1.0
+ */
+
+public class Graphics {
+
+    // SYNC NOTE: the main Graphics class is entirely unlocked.  There is only
+    // one instance of Graphics (created in Display) and it is only ever legal
+    // for the application to use it when the system calls the paint() method
+    // of a Canvas.  Since paint() calls are serialized, and these methods
+    // modify only instance state, no locking is necessary.  (If any method
+    // were to read or modify global state, locking would need to be done in
+    // those cases.)
+
+    /**
+     * Constant for centering text and images horizontally
+     * around the anchor point
+     *
+     * <P>Value <code>1</code> is assigned to <code>HCENTER</code>.</P>
+     */
+    public static final int HCENTER = 1;
+  
+    /**
+     * Constant for centering images vertically
+     * around the anchor point.
+     *
+     * <P>Value <code>2</code> is assigned to <code>VCENTER</code>.</P>
+     */
+    public static final int VCENTER = 2;
+  
+    /**
+     * Constant for positioning the anchor point of text and images
+     * to the left of the text or image.
+     *
+     * <P>Value <code>4</code> is assigned to <code>LEFT</code>.</P>
+     */
+    public static final int LEFT = 4;
+  
+    /**
+     * Constant for positioning the anchor point of text and images
+     * to the right of the text or image.
+     *
+     * <P>Value <code>8</code> is assigned to <code>RIGHT</code>.</P>
+     */
+    public static final int RIGHT = 8;
+  
+    /**
+     * Constant for positioning the anchor point of text and images
+     * above the text or image.
+     *
+     * <P>Value <code>16</code> is assigned to <code>TOP</code>.</P>
+     */
+    public static final int TOP = 16;
+  
+    /**
+     * Constant for positioning the anchor point of text and images
+     * below the text or image.
+     *
+     * <P>Value <code>32</code> is assigned to <code>BOTTOM</code>.</P>
+     */
+    public static final int BOTTOM = 32;
+  
+    /**
+     * Constant for positioning the anchor point at the baseline of text.
+     *
+     * <P>Value <code>64</code> is assigned to <code>BASELINE</code>.</P>
+     */
+    public static final int BASELINE = 64;
+
+    /**
+     * Constant for the <code>SOLID</code> stroke style.
+     *
+     * <P>Value <code>0</code> is assigned to <code>SOLID</code>.</P>
+     */
+    public static final int SOLID = 0;
+
+    /**
+     * Constant for the <code>DOTTED</code> stroke style.
+     *
+     * <P>Value <code>1</code> is assigned to <code>DOTTED</code>.</P>
+     */
+    public static final int DOTTED = 1;
+
+    private native void init0(Font font);
+    private native void initScreen0(int displayId, int width, int height);
+    private native void initImage0(Image img, int width, int height);
+
+    /**
+     * Create a Graphics object
+     */
+    Graphics() {
```

Comment:
> I think the constructor could be "nativized" as well (see for example, string.js)

We'd have to get the default font from somewhere. It's kind of frustrating that `getDefaultFont()` is `synchronized (Display.LCDUILock)` - it means that it's potentially much slower than it needs to be (due to locking) and also that we can't call it from JS (because it yields).